plugins/cache-caffeine/src/internalClusterTest/java/org/opensearch/cache/CaffeineHeapCacheIT.java

```diff
@@ -8,5 +8,268 @@
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public void testPluginsAreInstalled() {
+        internalCluster().startNode(Settings.builder().put(defaultSettings()).build());
+        NodesInfoRequest nodesInfoRequest = new NodesInfoRequest();
+        nodesInfoRequest.addMetric(NodesInfoRequest.Metric.PLUGINS.metricName());
+        NodesInfoResponse nodesInfoResponse = OpenSearchIntegTestCase.client().admin().cluster().nodesInfo(nodesInfoRequest).actionGet();
+        List<PluginInfo> pluginInfos = nodesInfoResponse.getNodes()
+            .stream()
+            .flatMap(
+                (Function<NodeInfo, Stream<PluginInfo>>) nodeInfo -> nodeInfo.getInfo(PluginsAndModules.class).getPluginInfos().stream()
+            )
+            .collect(Collectors.toList());
+        Assert.assertTrue(
+            pluginInfos.stream().anyMatch(pluginInfo -> pluginInfo.getName().equals("org.opensearch.cache.CaffeineCachePlugin"))
+        );
+    }
+
+    public void testSanityChecksWithIndicesRequestCache() throws InterruptedException {
+        internalCluster().startNode(Settings.builder().put(defaultSettings()).build());
+        Client client = client();
+        assertAcked(
+            client.admin()
+                .indices()
+                .prepareCreate("index")
+                .setMapping("f", "type=date")
+                .setSettings(
+                    Settings.builder()
+                        .put(IndexMetadata.SETTING_NUMBER_OF_SHARDS, 1)
+                        .put(IndexMetadata.SETTING_NUMBER_OF_REPLICAS, 0)
+                        .put(IndicesRequestCache.INDEX_CACHE_REQUEST_ENABLED_SETTING.getKey(), true)
+                        .build()
+                )
+                .get()
+        );
+        indexRandom(
+            true,
+            client.prepareIndex("index").setSource("f", "2014-03-10T00:00:00.000Z"),
+            client.prepareIndex("index").setSource("f", "2014-05-13T00:00:00.000Z")
+        );
+        ensureSearchable("index");
+
+        // This is not a random example: serialization with time zones writes shared strings
+        // which used to not work well with the query cache because of the handles stream output
+        // see #9500
+        final SearchResponse r1 = client.prepareSearch("index")
+            .setSize(0)
+            .setSearchType(SearchType.QUERY_THEN_FETCH)
+            .addAggregation(
+                dateHistogram("histo").field("f")
+                    .timeZone(ZoneId.of("+01:00"))
+                    .minDocCount(0)
+                    .dateHistogramInterval(DateHistogramInterval.MONTH)
+            )
+            .get();
+        assertSearchResponse(r1);
+
+        // The cached is actually used
+        assertThat(
+            client.admin().indices().prepareStats("index").setRequestCache(true).get().getTotal().getRequestCache().getMemorySizeInBytes(),
+            greaterThan(0L)
+        );
```

Comment:
Now that the new cache stats API is in place it'd be good to also check it works with your cache for all of these tests. Look at checkCacheStatsAPIResponse and testCacheStatsAPIWIthOnHeapCache in CacheStatsAPIIndicesRequestCacheIT.java for an example of how to do this (check for hits, misses, memory, eviction, items)

Reply:
(this is the API you invoke with `GET /_nodes/stats/caches/request_cache` during benchmarking, as opposed to the older one with `GET /_nodes/stats/indices/request_cache`)